src/tests/cmocka/data_provider/test_dp_request.c

```diff
@@ -312,6 +314,66 @@ static void test_type_mismatch(void **state)
     talloc_free(md);
 }
 
+static void test_permit_handler(void **state)
+{
+    errno_t ret;
+    struct test_ctx *test_ctx;
+    struct tevent_req *req;
+    struct dp_req_params params;
+    struct pam_data *pd;
+    struct pam_data *out_pd;
+
+    test_ctx = talloc_get_type(*state, struct test_ctx);
+
+    params.ev = test_ctx->tctx->ev;
+
```

Comment:
Can you create mock_dp_params in mock_dp.c and create the whole structure please? What if the handler changes and will want to touch something else then event context :-)